### ISSUE TITLE
Unknown compression table version

### ISSUE THREAD

**Issue by rrwright** [akka/akka#26786]
The symptoms previously reported/resolved in issue https://github.com/akka/akka/issues/23504 seem to still be occurring. Errors like the following appear many times in logs for the system described below:

`[WARN] [04/21/2019 12:33:12.912] [adapt-cluster-akka.remote.default-remote-dispatcher-265] [InboundActorRefCompression(akka://adapt-cluster)] Inbound message from originUid [-6777820092983884009] is using unknown compression table version. It may have been sent with compression table built for previous incarnation of this system. Versions activeTable: 83, nextTable: 84, incomingTable: 79`
`[WARN] [04/21/2019 12:33:12.912] [adapt-cluster-akka.remote.default-remote-dispatcher-265] [InboundActorRefCompression(akka://adapt-cluster)] Inbound message from originUid [-4623654363590644046] is using unknown compression table version. It may have been sent with compression table built for previous incarnation of this system. Versions activeTable: 75, nextTable: 76, incomingTable: 71`

(these are contiguous lines from our log)

The system producing the error is a 6-machine cluster using:
* Akka version: `2.5.22`
* Scala version: `2.12.8`
* using Akka artery with the `TCP` transport
* using akka packages: `akka-actor`, `akka-stream`, `akka-cluster`, `akka-http`, `akka-http-spray-json`, `akka-stream-kafka`
* Akka HTTP version: `10.1.7`
* Akka stream kafka version: `0.22`

The compression table issue appears in our log beginning immediately after several warnings about the heartbeat interval growing too large.

**Comment by patriknw** [comment]
@rrwright Thanks for reporting. Can you share the full logs from both nodes leading up to this problem (at least from 6 minutes before) .

Do you have a way to reproduce it and would you be able to run with `akka.loglevel=DEBUG` and share logs?

**Comment by rrwright** [comment]
The error is not reliably reproducible. We've been rerunning the application/data and retesting for several days and came across the error one more time. That log (with noise from our own log messages, apologies) is attached:
[akka-error.out.zip](https://github.com/akka/akka/files/3123347/akka-error.out.zip)

Unfortunately, we only collected the log from the cluster member that produced the observable error message, and not every from member of the cluster; sorry about that.

If I stumble upon a more reliable way to reproduce the error, I will run with `akka.loglevel=DEBUG` and post the new set of logs.

**Comment by patriknw** [comment]
Thanks @rrwright . This log entry indicates that something is very wrong with the environment or the application:
```
[WARN] [04/24/2019 12:29:59.465] [adapt-cluster-akka.actor.default-dispatcher-157] [akka.cluster.Cluster(akka://adapt-cluster)] Cluster Node [akka://adapt-cluster@10.0.6.43:2551] - Scheduled sending of heartbeat was delayed. Previous heartbeat was sent [71384] ms ago, expected interval is [4000] ms. This may cause failure detection to mark members as unreachable. The reason can be thread starvation, e.g. by running blocking tasks on the default dispatcher, CPU overload, or GC.
```

and there are more similar later with even longer delays

**Comment by rrwright** [comment]
Yes, this log was produced while testing system behavior in the presence of pathological garbage collection—GC pauses over a minute. Our application doesn't operate under those conditions, but we wanted to test with an understanding that situation; when we found a bug outside our code, we thought it best to report it.

I only saw the compression table errors in logs testing pathological GC pauses. If you feel that is too much of an edge case and not behavior that Akka is expected to perform under, feel free to close as a "won't fix" or with the appropriate reason. You guys would know best what you're trying to support. On the other hand, if there is an underlying cause you think is worth addressing, you might be able to reproduce it by inducing long GC delays. Apologies for not mentioning the pathological testing circumstances early on!

**Comment by patriknw** [comment]
Thanks, that is good to know. I’ll look into if I can reproduce it in that way and think about what we can do.

I renews the compression tables once per minute, and keeps a few around in flight. It’s not supposed to bump to next table unless it got confirmation that previous was received. Might be something that need hardening there.

**Comment by patriknw** [comment]
I was able to reproduce something like this with the attached sample app, and using `kill -STOP` and `kill -CONT`. Found that it's probably caused by https://github.com/akka/akka/issues/26910. The scheduler triggered changing of compression tables (phasing out the old ones) in quick succession when the the JVM process wakes up. Didn't find anything else that had to be changed when the scheduler issue has been fixed.

[SimpleClusterApp.scala.zip](https://github.com/akka/akka/files/3176278/SimpleClusterApp.scala.zip)

**Comment by jrudolph** [comment]
> Didn't find anything else that had to be changed when the scheduler issue has been fixed.

Wouldn't that still be race condition that could be handled more gracefully?

**Comment by patriknw** [comment]
> Wouldn't that still be race condition that could be handled more gracefully?

It will fix itself as soon as the new tables are announced and used (unless we have a bug in that part that I can't see).

The default announcement interval is 1 minute and we keep current and 3 old tables around. If there are no incoming messages there will be no new tables. I can't see that it can be a race condition in practice.

The problem the scheduler introduced was that it was running minutes of announcements rapidly when the process woke up and therefor discarding old tables while there still were incoming messages in flight.

I was thinking that we could publish current table immediately (again) if we receive a message using a too old table that we have discarded, but don't want to add additional complexity unless showed to be a real problem.

**Comment by jrudolph** [comment]
I see, thanks for the explanation.

**Comment by patriknw** [comment]
Fixed via #26910
